api/v3/PaymentProcessor.php

```diff
@@ -128,6 +128,11 @@ function _civicrm_api3_payment_processor_getlist_defaults(&$request) {
 function civicrm_api3_payment_processor_pay($params) {
   $processor = Civi\Payment\System::singleton()->getById($params['payment_processor_id']);
   $processor->setPaymentProcessor(civicrm_api3('PaymentProcessor', 'getsingle', ['id' => $params['payment_processor_id']]));
+
+  // Add some aliases that doPayment expects. These should go away once we've got getters for them in CiviCRM core.
+  // Historically doPayment has normally been passed 'contributionID'
+  $params['contributionID'] = $params['contribution_id'];
```

Comment:
This is part of the pay api - I'm just gonna focus on that one until it's merged since there is a bit of a jumble of things happening here. I'll also look at the new docs